Creating extension fails over missing bean `RepositorySystem`

### Describe the bug

Extension creation fails.

### Expected behavior

Success.

### Actual behavior

```
[mvavrik@fedora clirep]$ ./quarkus-dev-cli create extension extension-abc --stream=3.0
[ERROR] ❗  Extension creation failed, No matching bean available: type is interface org.eclipse.aether.RepositorySystem
[ERROR] ❗  Unable to create extension: No matching bean available: type is interface org.eclipse.aether.RepositorySystem
```


### How to Reproduce?

Reproducer:

```
cat <<EOF > ./quarkus-dev-cli
#!/bin/bash
java -jar /home/mvavrik/.m2/repository/io/quarkus/quarkus-cli/999-SNAPSHOT/quarkus-cli-999-SNAPSHOT-runner.jar "\$@"
EOF
```
```
chmod +x ./quarkus-dev-cli
```
```
./quarkus-dev-cli create extension extension-abc --stream=3.0
```


### Output of `uname -a` or `ver`

Linux

### Output of `java -version`

openjdk 17.0.4 2022-07-19

### GraalVM version (if different from Java)

OpenJDK Runtime Environment GraalVM CE 22.3

### Quarkus version or git rev

999-SNAPSHOT

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.8.6

### Additional information

I am having identical issues when creating app, I can provide reproducer if needed.

--- a/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java
+++ b/core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/JarResultBuildStep.java
@@ -114,7 +114,9 @@ public class JarResultBuildStep {
         @Override
         public boolean test(String path) {
             return "META-INF/io.netty.versions.properties".equals(path) ||
-                    (path.startsWith("META-INF/services/") && path.length() > 18);
+                    (path.startsWith("META-INF/services/") && path.length() > 18) ||
+            // needed to initialize the CLI bootstrap Maven resolver
+                    "META-INF/sisu/javax.inject.Named".equals(path);
         }
     };
 

--- a/core/runtime/pom.xml
+++ b/core/runtime/pom.xml
@@ -225,7 +225,7 @@
                         <excludedArtifact>javax.enterprise:cdi-api</excludedArtifact>
                         <excludedArtifact>org.jboss.spec.javax.annotation:jboss-annotations-api_1.2_spec</excludedArtifact>
                         <excludedArtifact>org.jboss.spec.javax.annotation:jboss-annotations-api_1.3_spec</excludedArtifact>
-                        <excludedArtifact>javax.inject:javax.inject</excludedArtifact>
+                        <!-- excludedArtifact>javax.inject:javax.inject</excludedArtifact needed to initialize the Maven resolver in the CLI -->
                         <excludedArtifact>org.jboss.spec.javax.interceptor:jboss-interceptors-api_1.2_spec</excludedArtifact>
                         <excludedArtifact>org.glassfish:javax.el</excludedArtifact>
                         <excludedArtifact>javax.annotation:javax.annotation-api</excludedArtifact>

--- a/independent-projects/bootstrap/maven-resolver/pom.xml
+++ b/independent-projects/bootstrap/maven-resolver/pom.xml
@@ -172,17 +172,6 @@
             </resource>
         </resources>
         <plugins>
-            <!-- <plugin>
-                <groupId>org.apache.maven.plugins</groupId>
-                <artifactId>maven-jar-plugin</artifactId>
-                <executions>
-                    <execution>
-                        <goals>
-                            <goal>test-jar</goal>
-                        </goals>
-                    </execution>
-                </executions>
-            </plugin> -->
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-surefire-plugin</artifactId>

